### PR TITLE
Some code clean up about annotations

### PR DESCRIPTION
**Pull request by siavashsoleymani**
Based on below issue i do some refactoring:
https://github.com/FAForever/downlords-faf-client/issues/1455

### CODE REVIEW

File: src/main/java/com/faforever/client/achievements/AchievementItemController.java [hunk 1]
```diff
@@ -37,7 +37,7 @@
   public ImageView imageView;
   private AchievementDefinition achievementDefinition;
 
-  @Inject
+
```
Comment (1-alex98) [inline src/main/java/com/faforever/client/achievements/AchievementItemController.java:40]: Remove line i guess then

File: src/main/java/com/faforever/client/api/dto/Mod.java [hunk 2]
```diff
@@ -35,4 +34,15 @@
 
   @Relationship("latestVersion")
   private ModVersion latestVersion;
+
```
Comment (1-alex98) [inline src/main/java/com/faforever/client/api/dto/Mod.java:37]: The other way arround add @AllArgsConstructor where possible

**Review by 1-alex98 — CHANGES_REQUESTED** [review]

File: src/main/java/com/faforever/client/achievements/AchievementItemController.java (hunk 1 above)
Comment (siavashsoleymani) [reply src/main/java/com/faforever/client/achievements/AchievementItemController.java:40]: Ok i will take care of it in a few days

File: src/main/java/com/faforever/client/api/dto/Mod.java (hunk 2 above)
Comment (siavashsoleymani) [reply src/main/java/com/faforever/client/api/dto/Mod.java:37]: Ok i will take care of it in a few days

**Pull request closed without merging**